src/vs/workbench/browser/parts/editor/editorsObserver.ts

```diff
@@ -359,9 +360,17 @@ export class EditorsObserver extends Disposable {
 
 			return true;
 		});
+		let editorsToCloseCount: number;
+		if (ignoreDirtyTabs === true) {
+			editorsToCloseCount = leastRecentlyClosableEditors.length - limit + 1;
```

Comment:
I am not sure this logic is entirely correct. The list of `leastRecentlyClosableEditors` is not only driven by dirty editors but also sticky editors (sticky editors are the "Pinned" editors in the UI). There is also another check for `never the editor that should be excluded`. I think the check for dealing with dirty editors or not needs to go somewhere closer where we check for dirty editors.

Reply:
@bpasero Thank you for your suggestions, I changed the logic about editorsToCloseCount , I feel it's more clear now， please review it.